steward/src/cellars/aave_v2_stablecoin.rs

```diff
@@ -1,77 +1,182 @@
             let call = ReinvestCall {
                 min_assets_out: params.min_assets_out.into(),
             };
-            AaveV2StablecoinCellarCalls::Reinvest(call).encode()
+            Ok(AaveV2StablecoinCellarCalls::Reinvest(call).encode())
         }
-        ClaimAndUnstakeAmount(params) => {
-            let call = ClaimAndUnstakeWithAmountCall {
-                amount: params.amount.into(),
+        Sweep(params) => {
+            let token = match params.token.parse::<H160>() {
+                Ok(t) => t,
+                Err(_) => {
+                    return Err(sp_call_error(format!(
+                        "{}: Sweep 'token': argument is an invalid address: {}",
+                        LOG_PREFIX, params.token
+                    )))
+                }
             };
-            AaveV2StablecoinCellarCalls::ClaimAndUnstakeWithAmount(call).encode()
-        }
-        ClaimAndUnstake(_) => {
-            let call = ClaimAndUnstakeCall {};
-            AaveV2StablecoinCellarCalls::ClaimAndUnstake(call).encode()
-        }
-        Rebalance(params) => {
-            let call = RebalanceCall {
-                new_lending_token: params
-                    .address
-                    .parse::<H160>()
-                    .expect("failed to parse token address"),
-                min_new_lending_token_amount: params.min_new_lending_token_amount.into(),
+            let to = match params.to.parse::<H160>() {
+                Ok(t) => t,
+                Err(_) => {
+                    return Err(sp_call_error(format!(
+                        "{}: Sweep 'to': argument is an invalid address: {}",
+                        LOG_PREFIX, params.to
+                    )))
+                }
             };
-            AaveV2StablecoinCellarCalls::Rebalance(call).encode()
-        }
-        AccruePlatformFees(_) => {
-            let call = AccruePlatformFeesCall {};
-            AaveV2StablecoinCellarCalls::AccruePlatformFees(call).encode()
+            let call = SweepCall { token, to };
+            Ok(AaveV2StablecoinCellarCalls::Sweep(call).encode())
         }
         TransferFees(_) => {
             let call = TransferFeesCall {};
-            AaveV2StablecoinCellarCalls::TransferFees(call).encode()
+            Ok(AaveV2StablecoinCellarCalls::TransferFees(call).encode())
         }
-        SetInputToken(params) => {
-            let call = SetInputTokenCall {
-                token: params
-                    .address
-                    .parse::<H160>()
-                    .expect("failed to parse token address"),
-                is_approved: params.is_approved,
-            };
-            AaveV2StablecoinCellarCalls::SetInputToken(call).encode()
+    }
+}
+
+fn validate_route(results: Vec<Result<H160, &String>>) -> Result<(), Error> {
+    let mut bad_addresses = String::new();
+    for r in results {
+        if let Err(addr) = r {
+            bad_addresses.push_str(&format!(", {}", addr))
```

Comment:
This method has the weird side effect of starting the line with a comma.

Reply:
It get's appending to `err_string` so the comma ends up making some sense. would prefer the first be a `:` but it still looks ok:

```bash
ERROR:
  Code: InvalidArgument
  Message: SP call error: AaveV2StablcoinCellar: Rebalance 'route' argument contained invalid address(s), 0x5FC8d32690cc91D4c39d9d3abcBD16989F8757071, 0x0000000000000000000000000000000000000
```